bridges/ZoneTelechargementBridge.php

```diff
@@ -0,0 +1,82 @@
+<?php
+class ZoneTelechargementBridge extends BridgeAbstract {
+	const NAME = 'Zone Telechargement';
+	const URI = 'https://ww2.zone-telechargement1.org/';
+	const DESCRIPTION = 'Suivi de série sur Zone Telechargement';
+	const MAINTAINER = 'sysadminstory';
+	const PARAMETERS = array(
+		'Suivre la publication des épisodes d\'une série en cours de diffusion' => array(
+			'url' => array(
+				'name' => 'URL de la série',
+				'type' => 'text',
+				'required' => true,
+				'title' => 'URL d\'une série sans le https://ww2.zone-telechargement1.org/',
+				'exampleValue' => 'telecharger-series/31079-halt-and-catch-fire-saison-4-french-hd720p.html'
+				)
+			)
+		);
+
+	public function collectData(){
+		$html = getSimpleHTMLDOM(self::URI . $this->getInput('url'))
+			or returnServerError('Could not request Zone Telechargement.');
+
+		// Get the TV show title
+		$qualityselector = 'div[style=font-size: 18px;margin: 10px auto;color:red;font-weight:bold;text-align:center;]';
+		$show = $html->find('div[style*=font-weight: bold;text-align: center;margin: 25px;]', 0)->plaintext;
+		$quality = explode("\n", $html->find($qualityselector, 0)->plaintext)[0];
+		$this->showTitle = $show . ' ' . $quality;
+
+		// Get the post content
+		$linkshtml = $html->find('div[class=postinfo]', 0);
+
+		$episodes = array();
+
+		$list = $linkshtml->find('a');
+		// Construct the tabble of episodes using the links
+		foreach($list as $element) {
```

Comment:
You could probably move this two loops, it would make it simpler ?

Reply:
On the page, links are listed by Hoster :
```
HosterA
Episode 1
Episode 2
Episode 3
Episode 4

HosterB
Episode 1
Episode 2
Episode 3
Episode 4
```

Actually, for each link, I check the Hoster name, and episode number, and then add to the two dimensional table episodes[episode number] a line with the actual link formatted as needed.

After collecting all the links in this table, I create the items by walking through the episode table : this loop is pretty simple, I think.

Merging the two loops in one would make me set multiple times the 'title' and 'uri' properties for each item : how would I know if this episode is already present without doing more lines and check that this two loops ?

I may be passing by a better way to do it, so I'm open to suggestion if you have one :)

Reply:
Oh, I didn't undestand that there was multiple content for a single element